sonar-server/time-travel-server.go

```diff
@@ -82,12 +84,56 @@ func (s *timeTravelServer) Start() {
 	log.Println("start timeTravelServer...")
 }
 
+func toLowerMap(m map[string]interface{}) {
+	for key, val := range m {
+		switch v := val.(type) {
+		case map[string]interface{}:
+			if e, ok := m[key].(map[string]interface{}); ok {
+				toLowerMap(e)
+				m[strings.ToLower(key)] = e
+				if strings.ToLower(key) != key {
+					delete(m, key)
+				}
+			} else {
+				log.Println("bad")
+			}
+
+		case []interface{}:
+			if e, ok := m[key].([]interface{}); ok {
+				for idx := range e {
+					switch e[idx].(type) {
+					case map[string]interface{}:
+						if ee, ok := e[idx].(map[string]interface{}); ok {
+							toLowerMap(ee)
+							e[idx] = ee
```

Comment:
This part might be tricky. After `ee, ok := e[idx].(map[string]interface{})`, is ee a reference to e[idx] or a deep copy? If it is a reference then there is no need to `e[idx] = ee` I guess?
Correct me if I am wrong.

Reply:
I think so, since the element we assert is a map, so it's taken by reference